ash/src/prelude.rs

```diff
@@ -19,3 +19,27 @@ impl vk::Result {
         }
     }
 }
+
+/// Repeatedly calls `f` until it does not return [`vk::Result::INCOMPLETE`] anymore,
+/// ensuring all available data has been read into the vector.
+///
+/// See for example [`vkEnumerateInstanceExtensionProperties`]: the number of available
+/// items may change between calls; if it has increased `INCOMPLETE` is returned because
+/// the result-vec is not large enough.
+///
+/// [`vkEnumerateInstanceExtensionProperties]: https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkEnumerateInstanceExtensionProperties.html
+pub(crate) unsafe fn read_into_vector<T>(
+    f: impl Fn(&mut u32, *mut T) -> vk::Result,
+) -> VkResult<Vec<T>> {
+    loop {
+        let mut count: u32 = 0;
+        f(&mut count, std::ptr::null_mut()).result()?;
+        let mut data = Vec::with_capacity(count as usize);
+
+        let err_code = f(&mut count, data.as_mut_ptr());
+        if err_code != vk::Result::INCOMPLETE {
+            data.set_len(count as usize);
+            break err_code.result_with_success(data);
```

Comment:
Style nit: no need to be cheeky here
```suggestion
            return err_code.result_with_success(data);
```

Reply:
`loop`-`break` is a common Rust pattern I'd like to embrace. If we shouldn't use `break <some_value>;` I don't think we should be using `loop` either and just convert this into `let mut foo = INCOMPLETE; while foo == INCOMPLETE` etc.

Reply:
No strong opinion.